lib/lint.js

```diff
@@ -0,0 +1,70 @@
+const { validateHubl } = require('@hubspot/cli-lib/api/validate');
+const { getPortalId } = require('@hubspot/cli-lib');
+const vscode = require('vscode');
+
+const getRange = (document, error) => {
+  if (error.startPosition > 0) {
+    return document.getWordRangeAtPosition(
+      new vscode.Position(error.lineno - 1, error.startPosition - 1)
+    );
+  } else {
+    return document.lineAt(new vscode.Position(error.lineno - 1, 0)).range;
+  }
+};
+
+async function updateDiagnostics(document, collection) {
+  const { renderingErrors } = await validateHubl(
+    getPortalId(),
+    document.getText()
+  );
+
+  const fitleredErrors = renderingErrors.filter((error) => {
+    return error.reason != 'MISSING';
+  });
+
+  const templateErrors = fitleredErrors.map((error) => {
+    return {
+      code: '',
+      message: error.message,
+      range: getRange(document, error),
+      severity: vscode.DiagnosticSeverity.Error,
+    };
+  });
+
+  if (document) {
```

Comment:
I'm pretty new to this vs code repo, but I think you can probably bail early in this function if document doesn't exist. It would save you from having to unnecessarily do the validateHubl, filter, and map above.

Reply:
AH, good catch. I'm trying to remember why I even added this, I feel like I might have intended to check something else. I'll take a second look.